fix: etcd RPC size limit exceeded when dropping collection

sijie-ni-0214 opened the pull request:
issue: https://github.com/milvus-io/milvus/issues/46410

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
- Core invariant: etcd metadata and in-memory Segment/TextIndex records must store only compact filenames for text-index files; full object keys are deterministically reconstructed at use-sites from a stable root + common.TextIndexPath + IDs via metautil.BuildTextLogPaths.

- Bug & fix (issue #46410): the etcd RPC size overflow was caused by persisting full upload keys in segment/TextIndex metadata. Fix: at upload/creation sites (internal/datanode/compactor/sort_compaction.go and internal/datanode/index/task_stats.go) store only filenames using metautil.ExtractTextLogFilenames; at consumption/use sites (internal/datacoord/garbage_collector.go, internal/querynodev2/segments/segment.go, and other GC/loader code) reconstruct full paths with metautil.BuildTextLogPaths before accessing object storage.

- Simplified/removed logic: removed the redundant practice of carrying full object keys through metadata and in-memory structures; callers now persist compact filenames and perform on-demand path reconstruction. This eliminates large payloads in etcd and reduces memory pressure while preserving the same runtime control flow and error handling.

- No data loss / no regression: filename extraction is a deterministic suffix operation (metautil.ExtractTextLogFilenames) and reloadFromKV performs backward compatibility (internal/datacoord/meta.go converts existing full-path entries to filenames before caching). All read paths reconstruct full paths at runtime (garbage_collector.getTextLogs, LocalSegment.LoadTextIndex, GC/loader), so no files are modified/deleted and access semantics remain identical.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

sijie-ni-0214 commented:
/ci-rerun-e2e-arm

bigsheeper commented on internal/querynodev2/segments/segment.go:
```diff
@@ -1189,11 +1191,22 @@ func (s *LocalSegment) LoadTextIndex(ctx context.Context, textLogs *datapb.TextI
+	fullPaths := make([]string, 0, len(textLogs.GetFiles()))
+	for _, filename := range textLogs.GetFiles() {
+		fullPaths = append(fullPaths, path.Join(prefix, filename))
+	}
```
better to wrap an  util function in metautil

sijie-ni-0214 committed:
fix: etcd RPC size limit exceeded when dropping collection

When dropping a collection, a large number of segments need to be marked for deletion.
At this point, the size of the RPC request written to etcd may exceed the limit, resulting in a panic.

Store only filenames in TextIndexStats.Files instead of full paths
to reduce memory and etcd storage overhead. Full paths are
reconstructed during loading and GC using segment metadata.

Signed-off-by: sijie-ni-0214 <sijie.ni@zilliz.com>

sijie-ni-0214 commented:
/ci-rerun-ut-integration

sijie-ni-0214 commented:
/ci-rerun-e2e-default

sijie-ni-0214 commented:
/ci-rerun-e2e-default

sijie-ni-0214 commented:
/ci-rerun-build

sijie-ni-0214 commented:
/ci-rerun-ut-go

czs007 reviewed (approved):
/lgtm
